packages/js/uri-resolvers/src/__tests__/package-to-wrapper-cache-resolver.spec.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  IUriResolutionContext,
+  Result,
+  Uri,
+  UriPackageOrWrapper,
+  UriResolutionContext,
+} from "@polywrap/core-js";
+import { expectHistory } from "./helpers/expectHistory";
+import { ClientConfigBuilder, defaultPackages, PolywrapClient } from "@polywrap/client-js";
+import { RecursiveResolver, StaticResolver, WrapperCache } from "@polywrap/uri-resolvers-js";
+import { ipfsPlugin } from "@polywrap/ipfs-plugin-js";
+import { ipfsResolverPlugin } from "@polywrap/ipfs-resolver-plugin-js";
+import { httpPlugin } from "@polywrap/http-plugin-js";
+import { httpResolverPlugin } from "@polywrap/http-resolver-plugin-js";
+import { ExtendableUriResolver } from "@polywrap/uri-resolver-extensions-js";
+import { defaultIpfsProviders } from "@polywrap/client-config-builder-js";
+import { PackageToWrapperCacheResolver } from "../../build";
+
+jest.setTimeout(200000);
+
+const getClientWithCacheResolver = (): PolywrapClient => {
+  const resolver = RecursiveResolver.from(
+    PackageToWrapperCacheResolver.from(
+      [
+        StaticResolver.from([
+          {
+            uri: defaultPackages.ipfs,
+            package: ipfsPlugin({}),
+          },
+          {
+            uri: defaultPackages.ipfsResolver,
+            package: ipfsResolverPlugin({}),
+          },
+          {
+            uri: defaultPackages.http,
+            package: httpPlugin({})
+          },
+          {
+            uri: defaultPackages.httpResolver,
+            package: httpResolverPlugin({})
+          }
+        ]),
+        new ExtendableUriResolver(),
+      ],
+      new WrapperCache()
+    )
+  );
+
+  const config = new ClientConfigBuilder(undefined, resolver)
+    .addEnv(
+      defaultPackages.ipfs,
+      {
+        provider: defaultIpfsProviders[0],
+        fallbackProviders: defaultIpfsProviders.slice(1),
+      })
+    .addInterfaceImplementations(
+      ExtendableUriResolver.extInterfaceUri,
+      [
+        defaultPackages.ipfsResolver,
+        defaultPackages.httpResolver,
+      ]
+    )
+    .buildCoreConfig();
+
+  return new PolywrapClient(config, { noDefaults: true });
+};
+
+describe("PackageToWrapperCacheResolver", () => {
+
+  it("repeated calls with same uri trigger only one network request", async () => {
```

Comment:
If we're using Mutexes, I think we should have tests that use workers to confirm it all works as expected.
Otherwise we could use Mutex-less request caching